crates/storage-query-datafusion/src/context.rs

```diff
@@ -99,86 +98,169 @@ pub trait SelectPartitions: Send + Sync + Debug + 'static {
     async fn get_live_partitions(&self) -> Result<Vec<(PartitionId, Partition)>, GenericError>;
 }
 
-#[derive(Clone)]
-pub struct QueryContext {
-    sql_options: SQLOptions,
-    datafusion_context: SessionContext,
+#[async_trait]
```

Comment:
Does this need to be a dyn-compatible trait? if not, can we remove async_trait?

Reply:
The only reason I made it object safe was in case we need to use the `QueryContext::from_iter` method [here](https://github.com/restatedev/restate/pull/2912/files/6fe80a5a8a5c6bf0aa3af7f59f56e284640f3c8b#diff-7e6ec2d2bfc1dca07b8e9052710eec40c25651aa15d497a6926c29bcbb9f0d54R223) 

But I think we can drop that until needed.